File: myapplication/src/main/java/com/flying/blur/MainActivity.java
package com.flying.blur;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.drawable.BitmapDrawable;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    View mainView;
    View mainView2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mainView = findViewById(R.id.main_view);
        mainView2 = findViewById(R.id.main_view2);
//        mainView.post(new Runnable() {
//            @Override
//            public void run() {
//                int mainViewHeight=mainView.getMeasuredHeight();
//                int mainViewWidth = mainView.getMeasuredWidth();
//                int mainView2Height= mainView2.getMeasuredHeight();
//                int mainView2Width = mainView2.getMeasuredWidth();
//                Bitmap bitmap1 = Bitmap.createBitmap(mainViewWidth,mainViewHeight, Bitmap.Config.ARGB_8888);
//                Bitmap bitmap2 = Bitmap.createBitmap(mainView2Width,mainView2Height,Bitmap.Config.ARGB_8888);
//                Canvas canvas = new Canvas(bitmap1);
//                canvas.drawColor(Color.CYAN);
//                Canvas canvas1 = new Canvas(bitmap2);
//                canvas1.drawColor(Color.CYAN);
//                mainView.setBackground(new BitmapDrawable(getResources(),bitmap1));
//                mainView2.setBackground(new BitmapDrawable(getResources(),BlurBitmapUtil.blurBitmap(getApplicationContext(),bitmap2,20)));
//            }
//        });
        mainView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.e("onClick","onClick");
                //Uri uri = Uri.parse("faceu://live/share");
                Uri uri = Uri.parse("faceu://faceu.zmxy.callback");
                Intent intent = new Intent();
                intent.setAction(Intent.ACTION_VIEW);
                intent.addCategory(Intent.CATEGORY_DEFAULT);
                intent.setData(uri);
                Log.e("onclick","schema:"+uri.getScheme()+"  host: "+uri.getHost()+" package: "+getPackageName());
                //intent.setPackage("com.lemon.faceu");
                startActivity(intent);
            }
        });

        mainView2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent();
                intent.setClass(MainActivity.this,MyService.class);
                startService(intent);
            }
        });
    }
}

File: app/src/main/java/com/slive/demo/lb/MainActivity.kt
package com.slive.demo.lb

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.slive.demo.R

class MainActivity : AppCompatActivity() {

//    val runnableLambda: ()->Unit = { android.util.Log.e("sliver","run")}
    val runnable = Runnable {
        android.util.Log.e("sliver","run")
    }
    val runnableObject = object:Runnable {
        override fun run() {
            android.util.Log.e("sliver","run")
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_test_run)
        initView()
    }

    private fun initView() {
        val addRunnable:TextView = findViewById(R.id.add_runnable)
        addRunnable.setOnClickListener{
            JavaRunQueue.addRunnable(runnable)
            JavaRunnable2.addRunnable(runnableObject)
            Log.e("sliver","JavaRunQueue.size: "+JavaRunQueue.size())
            Log.e("sliver","JavaRunnable2.size: "+ JavaRunnable2.size())
        }

        val removeRunnable: TextView = findViewById(R.id.remove_runnable)
        removeRunnable.setOnClickListener {
            JavaRunQueue.removeRunnable(runnable)
            JavaRunnable2.removeRunnable(runnableObject)
            Log.e("sliver","JavaRunQueue.size: "+JavaRunQueue.size())
            Log.e("sliver","JavaRunnable2.size: "+ JavaRunnable2.size())
        }
    }
}
